profile/profile.go

```diff
@@ -450,20 +450,16 @@ func (p *Profile) Aggregate(inlineFrame, function, filename, linenumber, address
 func (p *Profile) NumLabelUnits() (map[string]string, map[string][]string) {
 	numLabelUnits := map[string]string{}
 	ignoredUnits := map[string]map[string]bool{}
+	encounteredKeys := map[string]bool{}
```

Comment:
I am somewhat curious about the performance impact this additional map might have as the loop in this function is really tight being called for every sample in a profile, but I don't have any data. We should probably add a [benchmark test](https://golang.org/pkg/testing/#hdr-Benchmarks) for profile encoding / decoding.

Reply:
Adding the benchmark shouldn't be part of this PR.